List/src/main/java/com/list/Node.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
+ * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
+ */
+package com.list;
+
+import java.util.ArrayList;
+
+/**
+ *
+ * @author vitor
+ */
+abstract class Node {
```

Comment:
Since this is the abstract class, it would be very useful to have doc comments at the class level, as well as for each method.